src/resmom/linux/alps.c

```diff
@@ -4208,13 +4277,20 @@ alps_create_reserve_request(job *pjob, basil_request_reserve_t **req)
 
 		/*
 		 * Only match vnodes that begin with mpphost and have
-		 * a following "_<num>_<num>".
+		 * a following "_<num>_<num>" (when vnode_per_numa_node
+		 * is true) otherwise, just plain "_<num>".
 		 */
 		if (strncmp(vp->vn_vname, mpphost, len) != 0)
 			continue;
 		cp = &vp->vn_vname[len];
-		if (sscanf(cp, "_%ld_%d", &nid, &seg) != 2)
-			continue;
+
```

Comment:
Please add a comment here: The remainder of the vnode name must match"_<num>_<num>" when vnode_per_numa_node is enabled, or "_<num>" when disabled.